static_src/components/page_header.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import style from 'cloudgov-style/css/cloudgov-style.css';
+
+import createStyler from '../util/create_styler';
+import AppStore from '../stores/app_store';
+import OrgStore from '../stores/org_store';
+import { pageKinds } from '../constants';
+import PageMetadataStore from '../stores/page_metadata_store';
+import SpaceStore from '../stores/space_store';
+
+
+// Prefer title set by props or PageMetadataStore, then fallback to guessing by
+// page kind
+function getTitle(props) {
+  const pageMetadata = PageMetadataStore.current;
+  let title = props.title || pageMetadata.title;
+  if (title) {
+    return title;
+  }
+
+  // TODO this seems strange that the page header needs to know about specific
```

Comment:
One way to limit this: make a page title component for each org, space and app that inherit or use shared functionality. Then `main` decides which one to render, and they only need to know about one store each?